### ISSUE TITLE
Blood Magic sacrificial dagger deathphase error

### ISSUE THREAD

**Issue by 1434967947** [SpongePowered/SpongeForge#1601]
```
[22:07:49] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event
.tracking.CauseTracker.printIncorrectPhaseCompletion(CauseTracker.java:273) */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event
.tracking.CauseTracker.completePhase(CauseTracker.java:195)                 */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.event
.tracking.CauseTracker.switchToPhase(CauseTracker.java:166)                 */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.netwo
rk.PacketUtil.onProcessPacket(PacketUtil.java:152)                          */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.network.PacketTh
readUtil$1.redirect$onProcessPacket$zjk000(SourceFile:539)                  */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.network.PacketTh
readUtil$1.run(SourceFile:13)                                               */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     java.util.concurrent.Executors
$RunnableAdapter.call(Executors.java:511)                                   */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     java.util.concurrent.FutureTas
k.run(FutureTask.java:266)                                                  */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.util.Util.func_1
81617_a(SourceFile:46)                                                      */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     org.spongepowered.common.Spong
eImplHooks.onUtilRunTask(SpongeImplHooks.java:256)                          */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.Minecraft
Server.redirect$onRun$zhj000(MinecraftServer.java:3953)                     */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.Minecraft
Server.func_71190_q(MinecraftServer.java:679)                               */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.dedicated
.DedicatedServer.func_71190_q(DedicatedServer.java:384)                     */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.Minecraft
Server.func_71217_p(MinecraftServer.java:624)                               */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     net.minecraft.server.Minecraft
Server.run(MinecraftServer.java:482)                                        */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     java.lang.Thread.run(Thread.ja
va:745)                                                                     */
[22:07:49] [Server thread/ERROR] [Sponge]: /*  Phases Remaining:
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*   - Phase: org.spongepowered.commo
n.event.tracking.phase.entity.DeathPhase@3d017a33                           */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     Context:
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Source
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: EntityPlayerMP['zyh'
/81681439, l='plot', x=-65.50, y=67.00, z=66.50]                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: DamageSource
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: DamageSource{Name=bl
oodMagic, Type=custom}                                                      */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: CapturedBlocks
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: CapturedBlocksSuppli
er{Captured=0}                                                              */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: CapturedEntities
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: CapturedEntitiesSupp
lier{Captured=0}                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*   - Phase: org.spongepowered.commo
n.event.tracking.phase.packet.UseItemPacketState@59b13b59                   */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     Context:
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Source
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: EntityPlayerMP['zyh'
/81681439, l='plot', x=-65.50, y=67.00, z=66.50]                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: PacketPlayer
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: EntityPlayerMP['zyh'
/81681439, l='plot', x=-65.50, y=67.00, z=66.50]                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Packet
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: net.minecraft.networ
k.play.client.CPacketPlayerTryUseItem@6f6d03a9                              */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Cursor
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: SpongeItemStackSnaps
hot{itemType=ItemAir{Name=null}, count=0}                                   */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: IgnoringCreative
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: false
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: ItemUsed
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: 1xitem.bloodmagic.sa
crificialDagger.@0                                                          */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: CapturedEntities
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: CapturedEntitiesSupp
lier{Captured=0}                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: CapturedBlocks
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: CapturedBlocksSuppli
er{Captured=0}                                                              */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Owner
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: EntityPlayerMP['zyh'
/81681439, l='plot', x=-65.50, y=67.00, z=66.50]                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Notifier
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: EntityPlayerMP['zyh'
/81681439, l='plot', x=-65.50, y=67.00, z=66.50]                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*   - Phase: org.spongepowered.commo
n.event.tracking.phase.plugin.ScheduledTaskPhaseState@221b26ed              */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     Context:
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: Source
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: com.google.common.ut
il.concurrent.ListenableFutureTask@1d467999                                 */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: CapturedBlocks
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: CapturedBlocksSuppli
er{Captured=0}                                                              */
[22:07:49] [Server thread/ERROR] [Sponge]: /*     - Name: CapturedEntities
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*       Object: CapturedEntitiesSupp
lier{Captured=0}                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /*
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /* Minecraft : 1.11.2
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /* SpongeAPI : 6.1.0-SNAPSHOT-a15cc3e
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /* SpongeForge : 1.11.2-2315-6.1.0-BE
TA-2395                                                                     */
[22:07:49] [Server thread/ERROR] [Sponge]: /* Minecraft Forge : 13.20.0.2315
                                                                            */
[22:07:49] [Server thread/ERROR] [Sponge]: /************************************
*****************************************************************************/
> /*****************************************************************************
************************************/
/*                                          Completing incorrect phase
                                 */
/*******************************************************************************
**********************************/
/* Sponge's tracking system is very dependent on
                                 */
/* knowing whena change to any world takes place,
                                 */
/* however, we are attemptingto complete a "phase"
                                 */
/* other than the one we most recently entered.This
                                 */
/* is an error usually on Sponge's part, so a
                                 */
/* reportis required on the issue tracker on GitHub.
                                 */
/*******************************************************************************
**********************************/
/* Expected to exit phase: org.spongepowered.common.event.tracking.phase.packet.
UseItemPacketState@59b13b59      */
/* But instead found phase: org.spongepowered.common.event.tracking.phase.entity
.DeathPhase@3d017a33             */
/* StackTrace:
                                 */
/* java.lang.Exception: null
                                 */
/*     org.spongepowered.common.event.tracking.CauseTracker.printIncorrectPhaseC
ompletion(CauseTracker.java:273) */
/*     org.spongepowered.common.event.tracking.CauseTracker.completePhase(CauseT
racker.java:195)                 */
/*     org.spongepowered.common.event.tracking.CauseTracker.switchToPhase(CauseT
racker.java:166)                 */
/*     org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.ja
va:152)                          */
/*     net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$zjk000(
SourceFile:539)                  */
/*     net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13)
                                 */
/*     java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
                                 */
/*     java.util.concurrent.FutureTask.run(FutureTask.java:266)
                                 */
/*     net.minecraft.util.Util.func_181617_a(SourceFile:46)
                                 */
/*     org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.ja
va:256)                          */
/*     net.minecraft.server.MinecraftServer.redirect$onRun$zhj000(MinecraftServe
r.java:3953)                     */
/*     net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:67
9)                               */
/*     net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServ
er.java:384)                     */
/*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:62
4)                               */
/*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:482)
                                 */
/*     java.lang.Thread.run(Thread.java:745)
                                 */
/*  Phases Remaining:
                                 */
/*   - Phase: org.spongepowered.common.event.tracking.phase.entity.DeathPhase@3d
017a33                           */
/*     Context:
                                 */
/*     - Name: Source
                                 */
/*       Object: EntityPlayerMP['zyh'/81681439, l='plot', x=-65.50, y=67.00, z=6
6.50]                            */
/*     - Name: DamageSource
                                 */
/*       Object: DamageSource{Name=bloodMagic, Type=custom}
                                 */
/*     - Name: CapturedBlocks
                                 */
/*       Object: CapturedBlocksSupplier{Captured=0}
                                 */
/*     - Name: CapturedEntities
                                 */
/*       Object: CapturedEntitiesSupplier{Captured=0}
                                 */
/*   - Phase: org.spongepowered.common.event.tracking.phase.packet.UseItemPacket
State@59b13b59                   */
/*     Context:
                                 */
/*     - Name: Source
                                 */
/*       Object: EntityPlayerMP['zyh'/81681439, l='plot', x=-65.50, y=67.00, z=6
6.50]                            */
/*     - Name: PacketPlayer
                                 */
/*       Object: EntityPlayerMP['zyh'/81681439, l='plot', x=-65.50, y=67.00, z=6
6.50]                            */
/*     - Name: Packet
                                 */
/*       Object: net.minecraft.network.play.client.CPacketPlayerTryUseItem@6f6d0
3a9                              */
/*     - Name: Cursor
                                 */
/*       Object: SpongeItemStackSnapshot{itemType=ItemAir{Name=null}, count=0}
                                 */
/*     - Name: IgnoringCreative
                                 */
/*       Object: false
                                 */
/*     - Name: ItemUsed
                                 */
/*       Object: 1xitem.bloodmagic.sacrificialDagger.@0
                                 */
/*     - Name: CapturedEntities
                                 */
/*       Object: CapturedEntitiesSupplier{Captured=0}
                                 */
/*     - Name: CapturedBlocks
                                 */
/*       Object: CapturedBlocksSupplier{Captured=0}
                                 */
/*     - Name: Owner
                                 */
/*       Object: EntityPlayerMP['zyh'/81681439, l='plot', x=-65.50, y=67.00, z=6
6.50]                            */
/*     - Name: Notifier
                                 */
/*       Object: EntityPlayerMP['zyh'/81681439, l='plot', x=-65.50, y=67.00, z=6
6.50]                            */
/*   - Phase: org.spongepowered.common.event.tracking.phase.plugin.ScheduledTask
PhaseState@221b26ed              */
/*     Context:
                                 */
/*     - Name: Source
                                 */
/*       Object: com.google.common.util.concurrent.ListenableFutureTask@1d467999
                                 */
/*     - Name: CapturedBlocks
                                 */
/*       Object: CapturedBlocksSupplier{Captured=0}
                                 */
/*     - Name: CapturedEntities
                                 */
/*       Object: CapturedEntitiesSupplier{Captured=0}
                                 */
/*
                                 */
/* Minecraft : 1.11.2
                                 */
/* SpongeAPI : 6.1.0-SNAPSHOT-a15cc3e
                                 */
/* SpongeForge : 1.11.2-2315-6.1.0-BETA-2395
                                 */
/* Minecraft Forge : 13.20.0.2315
```
JAVA8
WINDOWS
 
- Plugins/Mods: 
[22:13:51] [Server thread/INFO]: Plugins (32): Minecraft, Minecraft Coder Pack,
Forge Mod Loader, Minecraft Forge, Micdoodle8 Core, SpongeAPI, SpongeForge, Baub
les, Guide-API, Blood Magic: Alchemical Wizardry, Botania, CustomNPCs, EconomyLi
te, IndustrialCraft 2, Galacticraft Core, Galacticraft Planets, LuckPerms, Nucle
us, PlotSquared, Project Core, WorldEdit, RedProtect, SlashBlade, Stack Ban, Pro
jectPortals, Holograms, StatusProtocol, Twenty Ticks, ProjectWorlds, Inspector,
keys, AdminShop
 

Unknowing crash problem

**Comment by axle2005** [comment]
Never mind that this is crazy hard to look at like this and SHOULD have been put in GIST. 

It seems your player killed himself using Blood magic, and the death didn't work properly. 

or: 

He killed a villager, which is already a known issue.

**Comment by 1434967947** [comment]
Sponge problem or Blood Magic problem?
if this is Sponge problem will it be fixed
by the way, sorry for not using GIST

**Comment by Zidane** [comment]
Closing this as its not comprehensible. Fix your logs and post a reply and I'll reopen. We also have a template for this, utilize it.

**Comment by FiveEYZ** [comment]
this is the error he is trying to report as i'm also getting it but for 1.10.2
https://gist.github.com/FiveEYZ/5597627cf73b8e288d6e38ea5236f8e4

A player was using a sacrificialDagger on himself until he gets half a heart, then he heals himself to full by eating. Later maybe 1-2 min (after he did a little crafting running around) he got back to use the sacrificialDagger again, he went from 20 hearts to half a heart on first click.

My theory is that something isn't updating the actual health of the player properly? so the player see that he is on 100% health, but the server sees the low health.

**Comment by Zidane** [comment]
@gabizou 

Thoughts?

**Comment by gabizou** [comment]
I feel like there was an exception before that one to have caused the `DeathPhase` to not have gone through. Would have to test either in dev or in production to reproduce. As far as the health is concerned, that seems more of a mod item issue as reviewing the synchronization bits for how health is managed on the server, it's not possible for the client to not have received any health updates until the first damage so to speak.

**Comment by phit** [comment]
@FiveEYZ @1434967947 is this still an issue with latest Blood Magic and Sponge?

**Comment by phit** [comment]
Closing this issue, as of today 1.10.2 and 1.11.2 are not longer supported. If a sufficiently, decent, and tested Pull Request is made to fix this bug/issue, it will still be considered, however, it is NOT to be expected that any bug fixes made for 1.12.2 will be back ported to 1.11.2 or 1.10.2.

If you can reproduce this issue with SpongeForge on Minecraft 1.12.2 and this issue was wrongly closed, please [open a new issue](https://github.com/SpongePowered/SpongeForge/issues/new) with updated information.
